src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -31,6 +31,52 @@ export function staticKeyToShadowsocksSessionConfig(staticKey: string): Shadowso
       password: config.password.data,
     };
   } catch (error) {
-    throw new errors.ServerUrlInvalid(error.message || 'Failed to parse static access key.');
+    throw new errors.ServerAccessKeyInvalid(error.message || 'Failed to parse static access key.');
+  }
+}
+
+// fetches information from a dynamic access key and attempts to parse it
+// TODO(daniellacosse): unit tests
+export async function promiseShadowsocksSessionConfigFromDynamicAccessKey(
+  dynamicKey: string
+): Promise<ShadowsocksSessionConfig> {
+  let response, sessionConfig;
+  try {
+    response = await fetch(dynamicKey);
+  } catch (error) {
+    throw new errors.ServerUnreachable(error.message || 'Failed to fetch VPN information from dynamic access key.');
```

Comment:
We should introduce a new error: FailedFetchConfig. ServerUnreachable os for the actual shadowsocks service, we need to differentiate them.